libs/react-spa/bff/src/lib/bff.hooks.ts

```diff
@@ -1,120 +1,55 @@
-    profile: {
-      sid: sid || '',
-      birthdate,
-      nationalId,
-      name,
-      idp,
-      actor,
-      subjectType,
-      delegationType,
-      locale,
-    },
-  }
-}
-
-/**
- * Dynamic hook to get the bff context.
+ * This hook is used to get the BFF authentication context.
  */
-export const useDynamicBffHook = (hookName: string): BffContextType => {
+export const useBff = () => {
   const bffContext = useContext(BffContext)
 
   if (!bffContext) {
-    throw new Error(`${hookName} must be used within a BffProvider`)
+    throw new Error('useBff must be used within a BffProvider')
   }
 
   return bffContext
 }
 
 /**
- * This hook is used to get the BFF authentication context.
- * It has backward compatibility with AuthContext.
+ * Dynamic hook to get specific key in the bff context.
  */
-export const useAuth = () => {
-  const bffContext = useContext(BffContext)
-  const authContext = useContext(AuthContext)
-
-  if (bffContext) {
-    return bffContext
-  }
+export const useDynamicBffHook = <Key extends keyof BffContextType>(
+  returnField: Key,
+): NonNullable<BffContextType[Key]> => {
+  const bffContext = useBff()
 
-  if (authContext) {
-    return authContext
+  if (!isDefined(bffContext[returnField])) {
+    throw new Error(`The field ${returnField} does not exist in the BffContext`)
   }
 
-  const errorMsg = (providerStr: string) =>
-    `useAuth must be used within a ${providerStr}`
-
-  if (!authContext) {
-    throw new Error(errorMsg('AuthProvider'))
-  }
-
-  throw new Error(errorMsg('BffProvider'))
+  return bffContext[returnField] as NonNullable<BffContextType[Key]>
 }
 
 /**
- * This hook is used to get user information.
- * It will determine what context to use based on the context that is available.
- * We will remove support for AuthContext when other clients transition over to BFF.
- * If AuthContext is being used then we will map the user info to the BffUser type.
+ * This hook is used to get the bff url generator.
+ * The bff url generator is used to generate urls for the Bff in a conveinent way.
  */
-export const useUserInfo = (): BffUser => {
-  const bffContext = useContext(BffContext)
-  const authContext = useContext(AuthContext)
-
-  const mappedAuthUserInfo = useMemo(() => {
-    if (authContext.userInfo) {
-      return mapToBffUser(authContext.userInfo)
-    }
+export const useBffUrlGenerator = () => useDynamicBffHook('bffUrlGenerator')
+export const useUserInfo = () => useDynamicBffHook('userInfo')
 
-    return null
-  }, [authContext.userInfo])
+export const useUserBirthday = () => {
+  const userInfo = useUserInfo()
 
-  if (bffContext?.userInfo) {
-    return bffContext.userInfo
-  } else if (mappedAuthUserInfo) {
-    return mappedAuthUserInfo
-  }
+  return useMemo(() => {
+    const nationalId = userInfo?.profile.nationalId
 
-  throw new Error('User info is not available. Is the user authenticated?')
+    return nationalId ? kennitala.info(nationalId)?.birthday : undefined
```

Comment:
This does not support kerfiskennitala users. We should read this from the ID token instead ... when the IDP adds support for it. 

Until then, we should probably add a comment/description to this hook to explain this limitation.